sdk/data/azcosmos/cosmos_dedicated_gateway_request_options.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import "time"
+
+// DedicatedGatewayRequestOptions includes options for operations in the dedicated gateway.
+type DedicatedGatewayRequestOptions struct {
```

Comment:
Added a `DedicatedGatewayRequestOptions` struct to match how dotnet uses the `MaxIntegratedCacheStaleness`: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/0fdb5e4222fad67ca6206b820d0d19e89fe4832c/Microsoft.Azure.Cosmos/src/Resource/Settings/DedicatedGatewayRequestOptions.cs#L14